packages/web-frontend/src/containers/Map/AreaTooltip.js

```diff
@@ -34,8 +41,76 @@ export class AreaTooltip extends Component {
     }
   }
 
+  getTextList() {
+    const { orgUnitName, hasMeasureValue, measureOptions, orgUnitMeasureData } = this.props;
+
+    const defaultTextList = [
+      <Heading key={0} hasMeasureValue={hasMeasureValue}>
+        {orgUnitName}
+      </Heading>,
+    ];
+
+    const renderTextList = (data, textList = []) => {
+      Object.keys(data).forEach(key => {
+        textList.push(<span key={key}>{`${key}: ${data[key]}`}</span>);
+      });
+      return textList;
+    };
+
+    if (hasMeasureValue) {
+      const legacy = measureOptions.length === 1;
+      const { formattedMeasureData, onlyOneValue } = legacy
+        ? this.buildLegacyFormattedMeasureData(orgUnitMeasureData, measureOptions)
+        : this.buildFormattedMeasureData(orgUnitMeasureData, measureOptions);
+
+      if (onlyOneValue) {
+        return renderTextList({ [orgUnitName]: formattedMeasureData });
+      }
+
+      renderTextList(formattedMeasureData, defaultTextList);
+    }
+
+    return defaultTextList;
```

Comment:
```suggestion
    const defaultTextList = [
      <Heading key={0} hasMeasureValue={hasMeasureValue}>
        {orgUnitName}
      </Heading>,
    ];
    
    if (!hasMeasureValue) {
      return defaultTextList;
    }
   
    const toTextList = (data) => Object.keys(data).map(key => <span key={key}>{`${key}: ${data[key]}`}</span>);
    
    if (measureOptions.length === 1) {
        return toTextList({ [orgUnitName]: getSingleFormattedValue(orgUnitMeasureData, measureOptions)});
    }

    return defaultTextList.concat(toTextList(this.buildFormattedMeasureData(orgUnitMeasureData, measureOptions)));
```

Feel this simplifies things a bit? Also means we can remove `buildLegacyFormattedMeasureData()` 👍

Reply:
@rohan-bes Nice refactor !👍